### PR TITLE
Disable reaction buttons when not logged in

### PR DESCRIPTION
**Pull request by Eik1234**
Closes webkom/lego#1861
Passes down loggedIn to Reaction component and disable clicking if not logged in.

### CODE REVIEW

File: app/components/Reactions/Reaction.js [hunk 1]
```diff
@@ -34,10 +35,11 @@ class Reaction extends React.Component<Props> {
       addReaction,
       deleteReaction,
       hasReacted,
+      canReact,
       reactionId,
       contentTarget,
     }: Props = this.props;
-    const classes = [className ? className : styles.reaction];
+    const classes = [className ? className : styles.reaction + (canReact ? " "+styles.clickable : "")];
```
Comment (LudvigHz) [inline app/components/Reactions/Reaction.js:42]: We use `classNames` here, hence the array. So it's better to add the classname as an element in the array.

A suggestion is to do something like this. (or something cleaner if you can :smile:)
```js
const classes = [className ? className : styles.reaction, canReact && styles.clickable].filter(Boolean) 
```

**Review by LudvigHz — APPROVED** [review]
Nice one! :100:
Users cannot actually access quotes when not logged in, but no harm in having the check there.

File: app/components/Reactions/Reaction.js [hunk 2]
```diff
@@ -34,10 +35,14 @@ class Reaction extends React.Component<Props> {
...
       deleteReaction,
       hasReacted,
+      canReact,
       reactionId,
       contentTarget,
     }: Props = this.props;
-    const classes = [className ? className : styles.reaction];
+    const classes = [
+      className ? className : styles.reaction,
+      ...(canReact ? [styles.clickable] : []),
```
Comment (SmithPeder) [inline app/components/Reactions/Reaction.js:44]: `classnames` can take false values https://www.npmjs.com/package/classnames, so you can actually don't need to pass the empty list if they cannot react. So you can actually do `canReact && styles.clickable`. No need for checks 👍

Comment (SmithPeder) [reply app/components/Reactions/Reaction.js:44]:
```js
// other falsy values are just ignored
classNames(null, false, 'bar', undefined, 0, 1, { baz: null }, ''); // => 'bar 1'
```

Comment (Eik1234) [reply app/components/Reactions/Reaction.js:44]: Ah, cool, i'll do that

**Review by SmithPeder — APPROVED** [review]
⭐

**Pull request merged**